Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreConfigurationRef.cpp

```diff
@@ -203,3 +203,31 @@ void WKWebsiteDataStoreConfigurationSetPCMMachServiceName(WKWebsiteDataStoreConf
 {
     WebKit::toImpl(configuration)->setPCMMachServiceName(name ? WebKit::toImpl(name)->string() : String());
 }
+
+double WKWebsiteDataStoreConfigurationGetOriginQuotaRatio(WKWebsiteDataStoreConfigurationRef configuration)
+{
+    auto result = WebKit::toImpl(configuration)->originQuotaRatio();
+    return result ? *result : -1.0;
+}
+
+void WKWebsiteDataStoreConfigurationSetOriginQuotaRatio(WKWebsiteDataStoreConfigurationRef configuration, double ratio)
+{
+    std::optional<double> value;
+    if (ratio != -1.0)
+        value = ratio;
+    WebKit::toImpl(configuration)->setOriginQuotaRatio(value);
+}
+
+double WKWebsiteDataStoreConfigurationGetTotalQuotaRatio(WKWebsiteDataStoreConfigurationRef configuration)
+{
+    auto result = WebKit::toImpl(configuration)->totalQuotaRatio();
+    return result ? *result : -1.0;
+}
+
+void WKWebsiteDataStoreConfigurationSetTotalQuotaRatio(WKWebsiteDataStoreConfigurationRef configuration, double ratio)
+{
+    std::optional<double> value;
+    if (ratio != -1.0)
+        value = ratio;
```

Comment:
Would it make more sense to add a WKWebsiteDataStoreConfigurationClearTotalQuotaRatio instead of using a special -1 value?

Reply:
I use a special value -1 as I think it is easy to make `WKWebsiteDataStoreConfigurationGetOriginQuotaRatio` return -1  to indicate the value is not set.

Does it help if I change it to const?